org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -434,9 +434,7 @@ private void resolve() {
 			return;
 		}
 		synchronized (this.javac) { // prevents from multiple `analyze` for the same task
-			boolean alreadyAnalyzed = this.converter.domToJavac.values().stream().map(TreeInfo::symbolFor).anyMatch(Objects::nonNull);
-			if (!alreadyAnalyzed) {
-				// symbols not already present: analyze
+			if (!this.converter.domToJavac.values().stream().map(TreeInfo::symbolFor).allMatch(Objects::nonNull)) {
```

Comment:
Do we have the guarantee that all DCTree have a matching symbol? For example, what about blocks or for/if/while....? I don't think this condition is viable. It may work because it recomputes all bindings greedily, but it might not be fine enough.
Moreover, if the symptom is that `converter.domToJavac` sometimes miss values at that stage, we should look at the root cause of why some content is missing here. It can be a synchronization issue, or a conversion issue, or ...?

Reply:
Do you have some case for which https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/blob/969bc5a07fb352fce857a4f954cdbdc6e8f7ec0f/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java#L640 is used? I'm not sure when it happens. I set a breakpoint there and I haven't been able to get it to hit with the AST conversion or completion test suites, nor when playing around in the IDE.

Reply:
This is only used when annotation processing is on (eg mapstruct or lombok or similar ones involved)

Reply:
Okay, thanks.

Reply:
> we should look at the root cause of why some content is missing here

I've been looking into this today. I identified one way to hack away the problem and pushed the code for that. The root cause is that one of the compilation units we process only has the last segment of the filepath. I'll need to keep looking to find the true root cause.

Reply:
Turns out the "last segment of the filepath" is the behaviour upstream as well. I have included a hack that should be able to prevent multiple versions of the same file from being included in the same compilation run, even when multiple files with the same name are open.